### PR TITLE
V22F-70 Intervention type menu

### PR DESCRIPTION
**Pull request by shoekzema**
Changes are in CHANGES.txt in the map src/simulatedDrivingAssistance

Added a new intervention type menu between the splash screen and the main menu in SD.

### CODE REVIEW

File: speed-dreams/source-2.2.3/data/data/menu/mainmenu.xml [hunk 1]
```diff
@@ -135,14 +135,37 @@
...
+		<attnum name="image x" val="-48"/>
+		<attnum name="image y" val="0"/>
+		<attnum name="image width" val="320"/>
+		<attnum name="image height" val="33"/>
+		<attstr name="focused image" val="data/img/bg-button-big-focused.png"/>
+		<attstr name="enabled image" val="data/img/bg-button-big.png"/>
+		<attstr name="pushed image" val="data/img/bg-button-big-pushed.png"/>
+	</section>
+		-->
+
```
[the remark this answers is not in the document]
Comment (shoekzema) [reply speed-dreams/source-2.2.3/data/data/menu/mainmenu.xml:160]: No, Ill remove it

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/legacymenu.cpp [hunk 2]
```diff
@@ -33,6 +33,9 @@
 #include <racemanagers.h>
 
 #include "splash.h"
+//SIMULATED DRIVING ASSISTANCE CHANGE:
```
[the remark this answers is not in the document]
Comment (shoekzema) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/legacymenu.cpp]: I don't think we want it, it already is noted in CHANGES.txt, I changed it

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 3]
```diff
@@ -71,6 +74,13 @@ onExitMenuActivate(void * /*dummy*/)
     GfuiScreenActivate(ExitMenuInit(MenuHandle));
 }
 
+// SIMULATED DRIVING ASSISTANCE CHANGE : commented out
+//static void OnInterventionMenuActivate(void * /*dummy*/)
+/*{
+    GfuiScreenActivate(InterventionMenuInit(MenuHandle));
+}*/
```
[the remark this answers is not in the document]
Comment (shoekzema) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp]: Done

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 4]
```diff
@@ -117,6 +127,8 @@ MainMenuInit(bool SupportsHumanDrivers)
     GfuiMenuCreateButtonControl(MenuHandle, menuDescHdle, "options", NULL, onOptionsMenuActivate);
     GfuiMenuCreateButtonControl(MenuHandle, menuDescHdle, "credits", NULL, onCreditsMenuActivate);
     GfuiMenuCreateButtonControl(MenuHandle, menuDescHdle, "quit", NULL, onExitMenuActivate);
+    //SIMULATED DRIVING ASSISTANCE CHANGE : button for intervention menu.
+    //GfuiMenuCreateButtonControl(MenuHandle, menuDescHdle, "intervention", NULL, OnInterventionMenuActivate);
```
[the remark this answers is not in the document]
Comment (shoekzema) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp]: Done

**Review by suppergerrie2 — APPROVED** [review]

**Review by LarsdeKwant — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/InterventionTests.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    ASSERT_NO_THROW((InterventionMakerIndication*)factory.CreateInterventionMaker(INTERVENTION_TYPE_INDICATION));
+    ASSERT_NO_THROW((InterventionMakerAskFor*)factory.CreateInterventionMaker(INTERVENTION_TYPE_ASK_FOR));
+    ASSERT_NO_THROW((InterventionMakerPerformWhenNeeded*)factory.CreateInterventionMaker(INTERVENTION_TYPE_PERFORM_WHEN_NEEDED));
+    ASSERT_NO_THROW((InterventionMakerAlwaysIntervene*)factory.CreateInterventionMaker(INTERVENTION_TYPE_ALWAYS_INTERVENE));
+}
+// Edge case
+TEST(FactoryTest, Edge)
+{
+    InterventionFactory factory;
+    ASSERT_NO_THROW((InterventionMakerNoIntervention*)factory.CreateInterventionMaker(NULL));
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/InterventionTests.cpp:70]: NULL is defined as 0. Which means this actually calls **factory.CreateInterventionMaker(0)**. This is equivalent to **factory.CreateInterventionMaker(INTERVENTION_TYPE_NO_INTERVENTION)**, meaning this test doesn't test anything it already has tested before. Instead you could test values of 6 and higher (since there are only 5 intervention types). It should still be able to cast to an InterventionMakerNoIntervention.

**Review by noutheesink — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]
